sites/geohub/src/components/controls/Modal.svelte

```diff
@@ -0,0 +1,74 @@
+<script lang="ts">
+  import { fade } from 'svelte/transition'
+  import { createEventDispatcher } from 'svelte'
+  import { Button } from '@undp-data/svelte-undp-design'
+  import Notification from '$components/controls/Notification.svelte'
+
+  const dispatch = createEventDispatcher()
+  export let dialogOpen = false
+  export let title
+  export let message
+  export let messageType: 'warning' | 'info' = 'warning'
+  export let target
+
+  const handleCancel = () => {
+    dispatch('cancel')
+  }
+  const handleKeyDown = (e) => {
+    if (e.key === 'Escape') {
+      handleCancel()
+    }
+  }
+  const handleContinue = () => {
+    dialogOpen = false
+    dispatch('continue')
+  }
+</script>
+
+<div
+  class="modal {dialogOpen ? 'is-active' : ''}"
+  data-testid="exit-confirmation-layer-view-container"
+  transition:fade>
+  <div
+    class="modal-background"
+    on:click={handleCancel}
+    on:keydown={handleKeyDown} />
+  <div class="modal-card">
+    <header class="modal-card-head">
+      <span class="modal-card-title">{title}</span>
+      <button
+        class="delete"
+        aria-label="close"
+        title="Close Delete Layer Button"
+        on:click={handleCancel} />
+    </header>
+    <section class="modal-card-body has-text-weight-normal">
+      <Notification
+        type={messageType}
+        showCloseButton={false}>
+        <div class="has-text-weight-medium">
+          {message}
+          <br />
+          <p>{target ? target : ''}</p>
+        </div>
+      </Notification>
+    </section>
+    <footer class="modal-card-foot is-flex is-flex-direction-row is-justify-content-flex-end">
+      <div class="footer-button px-2">
+        <Button
+          title="Cancel"
+          isPrimary={false}
+          on:clicked={handleCancel} />
+      </div>
+      <div class="footer-button px-2">
+        <Button
+          title="Continue"
+          isPrimary={true}
+          on:clicked={handleContinue} />
+      </div>
```

Comment:
can you use bulma button instead? I am using it in this component.

https://github.com/UNDP-Data/geohub/blob/6a3f814058e0806ba8e25e7e8fe1ef33dbac54c0/sites/geohub/src/components/data-upload/IngestingDatasets.svelte#L265-L277

bulma button has an option of `is-loading` style. I think this is quite nice feature. You can also export isLoading variable in the component.

https://bulma.io/documentation/elements/button/

also, you can export button names as well. so it will be flexible to change.

change button name to primary button / secondary button instead of using cancel / continue. primary button uses color of `is-primary`, secondary button uses color of `is-link`.